include/odp/api/spec/ipsec.h

```diff
@@ -1715,6 +1786,32 @@ int odp_ipsec_result(odp_ipsec_packet_result_t *result, odp_packet_t packet);
  */
 int odp_ipsec_status(odp_ipsec_status_t *status, odp_event_t event);
 
+/**
+ * IPSEC test API for modifying internal state of an SA.
+ *
+ * This function is not meant to be used by normal applications but by special
+ * test applications that test or debug the operation of the underlying ODP
+ * implementation. Calling this function may degrade the performance of the
+ * calling thread, other threads or the IPSEC implementation in general.
+ *
+ * Calling this function for an SA at the same time when the SA is used for
+ * processing traffic or when the SA is being modified through other parts
+ * of IPSEC API may result in undefined behaviour.
+ *
+ * SA state update through this function may not be supported by all ODP
+ * implementations, ODP instances or SA instances or at every moment. This
+ * function may fail for unspecified reasons even when the capability
+ * call indicated support for updating a particular parameter and previous
+ * similar calls succeeded.
+ *
+ * @param          sa            IPSEC SA to be updated
+ * @param          sa_param      IPSEC SA param to update
+ *
+ * @return 0      On success
+ * @retval <0     On failure
+ */
+int odp_ipsec_test_sa_update(odp_ipsec_sa_t sa, odp_ipsec_test_sa_param_t sa_param);
```

Comment:
I think Petri's comment on this got somehow lost, so I am repeating it: The ID of the parameter that is modified could be a parameter of this function instead of contained in the sa_param parameter. Then odp_ipsec_sa_test_param_t could itself be the union of possible values.

Reply:
Agreed.